Content.Shared/Zombies/SharedPendingZombieSystem.cs

```diff
@@ -0,0 +1,84 @@
+﻿using Content.Shared.Damage;
+using Content.Shared.Mobs;
+using Content.Shared.Mobs.Components;
+using Content.Shared.Mobs.Systems;
+using Content.Shared.Popups;
+using Robust.Shared.Timing;
+
+namespace Content.Shared.Zombies;
+
+/// <summary>
+///   Deals damage to bitten zombie victims each tick until they die. Then (serverside) zombifies them.
+/// </summary>
+public abstract class SharedPendingZombieSystem : EntitySystem
+{
+    [Dependency] private readonly IGameTiming _timing = default!;
+    [Dependency] private readonly DamageableSystem _damageable = default!;
+    [Dependency] private readonly SharedPopupSystem _popup = default!;
+    [Dependency] private readonly MobThresholdSystem _mobThreshold = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<PendingZombieComponent, EntityUnpausedEvent>(OnUnpause);
+    }
+
+    // Hurt them each second. Once they die, PendingZombieSystem will call Zombify and remove PendingZombieComponent
+    public override void Update(float frameTime)
+    {
+        base.Update(frameTime);
+        var curTime = _timing.CurTime;
+
+        var query = EntityQueryEnumerator<PendingZombieComponent, ZombieComponent, DamageableComponent, MobStateComponent>();
+        // Hurt the living infected
+        while (query.MoveNext(out var uid, out var pending, out var zombie, out var damage, out var mobState))
+        {
+            // Process only once per second
+            if (pending.NextTick > curTime)
+                continue;
+
+            // Don't hurt again for a while.
+            pending.NextTick += TimeSpan.FromSeconds(1);
+            if (pending.NextTick < curTime)
+            {
+                // If the component is far behind the current time, reinitialize it.
+                pending.NextTick = curTime + TimeSpan.FromSeconds(1);
+            }
```

Comment:
It should never be more than a second behind unless bugs occurred so they should be fixed instead.